indico/modules/events/editing/client/js/components/EditableSubmissionButton.jsx

```diff
@@ -0,0 +1,77 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+import editableURL from 'indico-url:event_editing.editable';
+import React, {useCallback, useState} from 'react';
+import PropTypes from 'prop-types';
+import {Button, Modal} from 'semantic-ui-react';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import {Translate} from 'indico/react/i18n';
+import FileManager from './FileManager';
+import {fileTypePropTypes} from './FileManager/util';
+
+export default function EditableSubmissionButton({
+  eventId,
+  contributionId,
+  uploadURL,
+  fileTypes,
+  type,
+  submitRevisionURL,
+}) {
+  const [submissionFiles, setSubmissionFiles] = useState({});
+  const [submitting, setSubmitting] = useState(false);
+  const submitRevision = async () => {
+    setSubmitting(true);
+    try {
+      await indicoAxios.put(submitRevisionURL, {
+        files: submissionFiles,
+      });
+    } catch (e) {
+      handleAxiosError(e, false, true);
+      setSubmitting(false);
+      return;
+    }
+    setSubmitting(false);
+    window.location.href = editableURL({confId: eventId, contrib_id: contributionId, type});
```

Comment:
I just noticed that we use `location.href` instead of `window.location.href` in most other places.